conans/client/generators/cmake_common.py

```diff
@@ -242,7 +242,7 @@ def generate_targets_section(dependencies):
                                        OR "${CONAN_CMAKE_CXX_STANDARD}" STREQUAL "gnu20")
     if(CONAN_STD_CXX_FLAG)
       message(STATUS "Conan setting CXX_FLAGS flags: ${CONAN_STD_CXX_FLAG}")
-      set(CMAKE_CXX_FLAGS "-std=${CONAN_STD_CXX_FLAG} ${CMAKE_CXX_FLAGS}")
+      set(CMAKE_CXX_FLAGS "${CONAN_STD_CXX_FLAG} ${CMAKE_CXX_FLAGS}")
```

Comment:
Maybe this error should be covered by a test.

Reply:
We don't have `cmake < 3` nor `gcc 8` in the CI. The max I can do is to check that the "-std=" is not there, but it is a ridiculous test.

Reply:
Added test in preflight test suite with a BR to cmake 2.8.12